### DISCUSSION TITLE
Issues in fuelcostprep.py and pcitures in README.md section in git

### DISCUSSION THREAD

**Discussion by balasubramaniansa** [NREL/ReEDS-2.0#136] [Q&A: Execution]
Hi ReEDS Team,
I try to run the ReEDS model (cases.csv & cases_test.csv). However both the runs ended in a few seconds. 

1. When I checked the output folder I can see an empty folder named (tc_phaseout_data).
2. When I checked the gamslog.txt file it showed the following information and stopped working.. 

Starting calculation of fuelcostprep.py
fuelcostprep.py completed successfully

Further in the installation README.md section in git the pictures are not visible and accessible.
Can you please help?

Thanks

**Comment by wesleyjcole** [comment]
I just pushed a commit to main that fixes the issue with the images not displaying.  You should be able to see them now.

writecapdat.py is the next input_processing file that is run after fuelcostprep.py (see the call_*.bat file in your run folder to see how the calls to the Python preprocessing files are made).  I suspect that you do not have one of the packages installed that writecapdat.py is using.  Would you check to make sure that Python can import the 5 packages at the top fo the writecapdat.py file?

**Reply by balasubramaniansa** [reply]
Thank you for your reply. I got the following error.

Unable to load gdxcc with default GAMS directory 'None'. You may need to explicitly call gdxpds.load_gdxcc(gams_dir) before importing pandas to avoid a library conflict.

Can you please help me with that? Thanks

**Reply by wesleyjcole** [reply]
Is GAMS installed on your machine?  If you are missing the gams_dir, you can try passing it directly in any gdxpds instances that might need it.  See http://nrel.github.io/gdx-pandas/api/gdxpds.html.

**Reply by balasubramaniansa** [reply]
Yes GAMS 30.3, Python 3.6.5, R 3.4.4 are installed in my machine. I am able to open the GAMS files (inputs, supply model, supply objective etc.) in my machine. I also installed GAMS 37 and 40 but in the ReEDS directory it shows GAMS release 30.3.0. 
The environment variables are set to GAMS 30.3 version. I have tried for example in writecapdat.py after import gdxpds I have added 
gdxpds.load_gdxcc(**GAMS dir**). This **GAMS dir** is the same directory given in path environment variables. I am getting the following error. I have attached the batch file in txt format.
[call_a9_ercot_seq.txt](https://github.com/NREL/ReEDS_OpenAccess/files/10397261/call_a9_ercot_seq.txt)

**Reply by wesleyjcole** [reply]
I'm not seeing the error you are getting.  It looks like it didn't make it in your post.

**Reply by balasubramaniansa** [reply]
This is the error I am getting. Thanks
[gamslog.txt](https://github.com/NREL/ReEDS_OpenAccess/files/10404827/gamslog.txt)

**Reply by wesleyjcole** [reply]
Your error is showing that writecapday.py is now working.  The new error is in cfgather.py, so if you implemented the same solution there, it should get that script working.  There's likely a more elegant fix here for addressing the GAMS_dir issue, but I don't know how to tackle that one.

**Reply by balasubramaniansa** [reply]
Actually I tried passing the GAMS_dir specifically in all the python files. 
also I tried running the model without specifically passing the Gams_dir in all the python files. 
In both the instances I am getting the same error as I attached previously (gamslog.txt).
i

**Reply by wesleyjcole** [reply]
I think ti might work if you passed in gams_dir when it calls a gdxpds function.  Your error files shows that the model fails on this line:

dupv = gdxpds.to_dataframe(gdxfile,'CFDUPV')

So adding a gams_dir into that function call will likely allow it to progress past that error.

**Reply by balasubramaniansa** [reply]
My Gams is stored in C:\GAMS\41. 
Can you please let me know how to add this gams_dir (C:\GAMS\41) into the function call. 
I tried various options but none of them worked.

**Reply by wesleyjcole** [reply]
The gxpds package is outside of where our expertise lies.  There is documentation for the package at http://nrel.github.io/gdx-pandas/api/gdxpds.html, but I'm not sure what next steps to take here.  I think that once the gdxpds call is resolved in the function, the model will progress through all the inputs scripts without issue.
